src/main/java/ch/uzh/ifi/hase/soprafs24/controller/UserController.java

```diff
@@ -44,4 +44,25 @@ public UserGetDTO createUser(@RequestBody UserPostDTO userPostDTO) {
     // convert internal representation of user back to API
     return DTOMapper.INSTANCE.convertEntityToUserGetDTO(createdUser);
   }
+
+  @PutMapping("/users/{id}")
+  @ResponseStatus(HttpStatus.OK)
+  @ResponseBody
+  public UserGetDTO updateUser(@PathVariable Long id, @RequestBody UserPostDTO userPostDTO) {
+    // convert API user to internal representation
+    User userInput = DTOMapper.INSTANCE.convertUserPostDTOtoEntity(userPostDTO);
+    userInput.setUserId(id);
+
+    // update user
+    User updatedUser = userService.updateUser(userInput);
+
+    // convert internal representation of user back to API
+    return DTOMapper.INSTANCE.convertEntityToUserGetDTO(updatedUser);
+  }
+
+  @DeleteMapping("/users/{id}")
+  @ResponseStatus(HttpStatus.OK)
+  public void deleteUser(@PathVariable Long id) {
+    userService.deleteUser(id);
```

Comment:
add auth check if auth token is user's token

Reply:
the code line below has been inserted to check for the auth token, if it's indeed the user's token

```
// Check if the token is valid and belongs to the user
    authorizationService.isAuthorized(token);
```

Reply:
same as above

Reply:
ok